### DISCUSSION TITLE
[iOS18] ipsw extract is not working due to Apple Encrypted Archive

### DISCUSSION THREAD

**Discussion by triceratops-with-a-single-horn** [blacktop/ipsw#490] [Q&A] [answered]
Apple released iOS 18 beta IPSWs today. When I try to run `ipsw extract` it's failing when it tries to mount the image.

Looking at the file structure it looks like some of the DMG files have `.aea` extension, which indicates Apple Encrypted Archive per [this wiki](https://www.theiphonewiki.com/wiki/Apple_Encrypted_Archive)
```
➜ ls ./iPhone14,2_18.0_22A5282m_Restore
090-27569-036.dmg.aea        090-28378-068.dmg            BuildManifest.plist          RestoreVersion.plist
090-27766-071.dmg            090-28659-047.dmg            Firmware                     SystemVersion.plist
090-27884-068.dmg            090-28659-047.dmg.aea        Restore.plist                kernelcache.release.iphone14
```

I tried to decrypt the file using aea command but it's (not surprisingly) giving "-key or -key-value is required" error. 
```
➜ aea decrypt -i 090-28659-047.dmg.aea -o out.dmg
-key or -key-value is required
```
and trying random keys (again not surprisingly) is not working
```
➜ aea decrypt -i 090-28659-047.dmg.aea -o out.dmg -key ./key
Failed to unlock container
Error 0xcf058400
Main key derivation failed (-1001)
Main key derivation
unlock container
```

**Steps to reproduce**
* Download any of the iOS 18 beta (22A5282m) IPSW files from Apple's downloads [page](https://developer.apple.com/download/) such as [this one](https://updates.cdn-apple.com/2024SummerSeed/fullrestores/052-34639/9BDA9FBA-6C1D-4702-9619-5CED4877EA5C/iPhone14,2_18.0_22A5282m_Restore.ipsw)
* Run `ipsw extract /path/to/downloaded/IPSW -d`
* Notice the error message `⨯ failed to IPSW FS dmg: failed to mount 090-28659-047.dmg.aea: exit status 1: hdiutil: attach failed - image not recognized`

**ipsw version**
Version: 3.1.492, BuildCommit: 0e84bf870f5398079eae84bd29cda2be0f1097d1

**Comment by blacktop** [comment, marked as answer]
Try latest release.  Let me know if there's still issues (requires to be ran on darwin due to `aea` bin requirement (for now))

**Reply by blacktop** [reply]
darwin no longer a requirement.
